src/App.jsx

```diff
@@ -14,70 +16,75 @@ export const goods = [
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood
+          ? (
+            <>
+              {`${selectedGood} is selected`}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => setSelectedGood('')}
+              />
+            </>
+          )
+          : 'No goods selected'}
+      </h1>
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const isSelectedGood = selectedGood === good;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+            return (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={
+                  cn({ 'has-background-success-light': isSelectedGood })
+                }
+              >
+                <td>
+                  <button
+                    data-cy={
+                      isSelectedGood
+                        ? 'RemoveButton'
+                        : 'AddButton'
+                    }
+                    type="button"
+                    className={
+                      cn('button', { 'is-info': isSelectedGood })
+                    }
+                    onClick={
+                      isSelectedGood
+                        ? () => setSelectedGood('')
+                        : () => setSelectedGood(good)
```

Comment:
Can we simplify it like this?
```suggestion
                         () => setSelectedGood(isSelectedGood ? '' :  good)
```